File: skeleton/src/main/java/me/shkschneider/skeleton/helpers/BroadcastHelper.java
package me.shkschneider.skeleton.helpers;

import android.content.BroadcastReceiver;
import android.content.Intent;
import android.content.IntentFilter;
import android.support.annotation.NonNull;

import me.shkschneider.skeleton.SkeletonApplication;

public class BroadcastHelper {

    public static boolean register(@NonNull final BroadcastReceiver broadcastReceiver, @NonNull final IntentFilter intentFilter) {
        SkeletonApplication.CONTEXT.registerReceiver(broadcastReceiver, intentFilter);
        return true;
    }

    public static boolean unregister(@NonNull final BroadcastReceiver broadcastReceiver) {
        SkeletonApplication.CONTEXT.unregisterReceiver(broadcastReceiver);
        return true;
    }

    public static boolean send(@NonNull final Intent intent) {
        SkeletonApplication.CONTEXT.sendBroadcast(intent);
        return true;
    }

}

File: android/src/main/kotlin/me/shkschneider/skeleton/android/content/BroadcastHelper.kt
package me.shkschneider.skeleton.android.content

import android.content.BroadcastReceiver
import android.content.Intent
import android.content.IntentFilter
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import me.shkschneider.skeleton.android.log.Logger
import me.shkschneider.skeleton.android.provider.ContextProvider
import me.shkschneider.skeleton.kotlin.jvm.tryOrNull

object BroadcastHelper {

    fun register(broadcastReceiver: BroadcastReceiver, vararg intentFilters: IntentFilter) {
        for (intentFilter in intentFilters) {
            LocalBroadcastManager.getInstance(ContextProvider.applicationContext()).registerReceiver(broadcastReceiver, intentFilter)
        }
    }

    fun send(intent: Intent) {
        LocalBroadcastManager.getInstance(ContextProvider.applicationContext()).sendBroadcast(intent)
    }

    // <https://stackoverflow.com/a/3568906>
    fun unregister(broadcastReceiver: BroadcastReceiver) {
        tryOrNull {
            LocalBroadcastManager.getInstance(ContextProvider.applicationContext()).unregisterReceiver(broadcastReceiver)
        }
    }

}
